generic3g/FieldUtilities.F90

```diff
@@ -0,0 +1,131 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_FieldUtilities
+   use MAPL_FieldPointerUtilities, only: FieldGetLocalElementCount
+   use mapl_ErrorHandling
+
+   use esmf
+   implicit none
+   private
+
+   public :: MAPL_FieldReallocate
+
+   interface MAPL_FieldReallocate
+      procedure :: reallocate
+   end interface MAPL_FieldReallocate
+
+   interface operator(==)
+      procedure :: ESMF_GeomEqual
+   end interface operator(==)
+
+   interface operator(/=)
+      procedure :: ESMF_GeomNotEqual
+   end interface operator(/=)
+
+contains
+
+
+   subroutine reallocate(field, geom, typekind, ungriddedUBound, rc)
```

Comment:
Does this subroutine reallocate the field empty, or does it reallocate the field with the data converted by typekind, shape, and geom? It looks like the former, but I want to confirm?

Reply:
Has to be the former.   For geom you would need to do a regrid.  For a new shape - where would you get the values?  typekind could be done.

The idea is that the reallocate is done inside of a coupler.   The new field will obtain new values when the update() step happens.

Reply:
OK